config/appstructure.go

```diff
@@ -0,0 +1,929 @@
+		for _, measure := range measures {
+			if measure == nil {
+				continue
+			}
+			obj.Measures = append(obj.Measures, AppStructureMeasureMeta{
+				LibraryId: measure.LibraryId,
+				Label:     measure.Def.Label,
+				Def:       measure.Def.Def,
+			})
+		}
+	}
+
+	rawListObject, _ := listObjects.Lookup(properties)
+	if rawListObject != nil {
+		var listObject enigma.ListObjectDef
+		if err := jsonit.Unmarshal(rawListObject, &listObject); err != nil {
+			return errors.WithStack(err)
+		}
+		obj.Dimensions = []AppStructureDimensionMeta{
+			{
+				LibraryId:       listObject.LibraryId,
+				LabelExpression: listObject.Def.LabelExpression,
+				Defs:            listObject.Def.FieldDefs,
+				Labels:          listObject.Def.FieldLabels,
+			},
+		}
+
+		if obj.Measures == nil {
+			obj.Measures = make([]AppStructureMeasureMeta, 0, len(listObject.Expressions))
+		}
+
+		for _, expression := range listObject.Expressions {
+			obj.Measures = append(obj.Measures, AppStructureMeasureMeta{
+				LibraryId: expression.LibraryId,
+				Def:       expression.Expr,
+			})
+		}
+	}
+
+	if obj.Selectable && (len(obj.Dimensions)+len(obj.Measures)) < 1 {
+		// object defined as selectable both doesn't have any data definitions found
+		structure.warn(fmt.Sprintf("object<%s> visualization<%s> type<%s> is expected to have data, but no measures or dimensions were found", obj.Id, vis, typ))
+	}
+
+	// no dimension = not selectable
+	if len(obj.Dimensions) < 1 {
+		obj.Selectable = false
+	}
+
+	return nil
+}
+
+func (structure *AppStructure) handleMeasure(ctx context.Context, app *senseobjects.App, id string, obj *AppStructureObject) error {
+	genMeasure, err := app.Doc.GetMeasure(ctx, id)
+	if err != nil {
+		fmt.Printf("Measure: %+v\n", genMeasure)
+		return errors.WithStack(err)
+	}
+	obj.RawBaseProperties, err = genMeasure.GetPropertiesRaw(ctx)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Save measure information to structure
+	var measure enigma.NxInlineMeasureDef
+	measurePath := senseobjdef.NewDataPath("/qMeasure")
+	rawMeasure, err := measurePath.Lookup(obj.RawBaseProperties)
+	if err != nil {
+		structure.warn(fmt.Sprintf("measure<%s> definition not found", id))
+	} else {
+		if err := jsonit.Unmarshal(rawMeasure, &measure); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	// Save meta information to structure
+	var meta MetaDef
+	metaPath := senseobjdef.NewDataPath("/qMetaDef")
+	rawMeta, err := metaPath.Lookup(obj.RawBaseProperties)
+	if err != nil {
+		structure.warn(fmt.Sprintf("measure<%s> has not meta information", id))
+	} else {
+		if err := jsonit.Unmarshal(rawMeta, &meta); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	obj.Measures = []AppStructureMeasureMeta{
+		{
+			Meta:  &meta,
+			Label: measure.Label,
+			Def:   measure.Def,
+		},
+	}
+
+	return nil
+}
+
+func (structure *AppStructure) handleDimension(ctx context.Context, app *senseobjects.App, id string, obj *AppStructureObject) error {
```

Comment:
Kind of a bad function name - bit too generic

Reply:
it an object with the specific type `dimension`, usually a dimension saved to master objects, but guess it could technically exist anyway. What would you suggest?